### PR TITLE
Refactor as a package

### PR DESCRIPTION
**Pull request by sarahgibs**
This is a big change. It allows the code in scan.py to be pulled in as a package, and treated as a proper dependency. This allows our actions that use this code to do so in a manner that makes just a little more sense.

### CODE REVIEW

File: examples/launcher.sh
Comment (TomConner) [inline examples/launcher.sh:1]: Minor: launcher.sh references `/zap` -- if this path is a prerequisite, consider adding a comment about how to do it (e.g. dockerfile with that as mount point or installing SDK at `/zap` or whatever you recommend)

**Review by TomConner — APPROVED** [review]
Several comments but none blockers. This is a great refactoring.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-python-dotenv==0.20.0
+python-dotenv==1.0.0
 selenium==4.1.5
 codedx-api @ git+https://github.com/broadinstitute/dsp-appsec-codedx-api-client-python.git@cf0de6fcdd0a4db98f2c2a35bd9d2ae7e8056d6b
 defectdojo_api @ git+https://github.com/DefectDojo/defectdojo_api#master
-google-cloud-storage==1.38.0
+google-cloud-storage==2.11.0
```
[the remark this answers is not in the document]
Comment (sarahgibs) [reply requirements.txt:5]: Good catch. They are no longer needed.

File: examples/reports/REPORTS.md
[the remark this answers is not in the document]
Comment (sarahgibs) [reply examples/reports/REPORTS.md:1]: This is required to run it, so I'm going to keep it here for now. I need to flesh out the examples folder, but it's not a requirement for getting it working with the GitHub Actions repo we use.

File: examples/launcher.sh
Comment (sarahgibs) [reply examples/launcher.sh:1]: The launcher runs in the zap docker image. And yeah there needs to be more explanation here.

**Pull request merged**